src/lists.rs

```diff
@@ -130,24 +159,50 @@ pub fn write_list<'b>(items: &[(String, String)], formatting: &ListFormatting<'b
             _ => {}
         }
 
-        result.push_str(item);
+        // Pre-comments
+        if let Some(ref comment) = item.pre_comment {
+            result.push_str(&rewrite_comment(comment,
+                                             false,
+                                             1000,
+                                             formatting.indent));
+            result.push('\n');
+            result.push_str(indent_str);
+        }
 
-        if tactic != ListTactic::Vertical && comment.len() > 0 {
-            if !comment.starts_with('\n') {
-                result.push(' ');
-            }
-            result.push_str(comment);
+        result.push_str(&item.item);
+
+        // Post-comments
+        if tactic != ListTactic::Vertical && item.post_comment.is_some() {
+            // We'll assume it'll fit on one line at this point
+            let formatted_comment = rewrite_comment(item.post_comment.as_ref().unwrap(),
+                                                    true,
+                                                    1000,
+                                                    0);
+
+            result.push(' ');
+            result.push_str(&formatted_comment);
         }
 
         if separate {
             result.push_str(formatting.separator);
         }
 
-        if tactic == ListTactic::Vertical && comment.len() > 0 {
-            if !comment.starts_with('\n') {
-                result.push(' ');
-            }
-            result.push_str(comment);
+        if tactic == ListTactic::Vertical && item.post_comment.is_some() {
+            let width = formatting.v_width - item_width - formatting.separator.len();
+            let offset = formatting.indent + item_width + formatting.separator.len();
+            let comment = item.post_comment.as_ref().unwrap();
+            // Use block-style only for the last item or multiline comments
```

Comment:
Yes, indeed. We have thus at least three options :
1. use bloc comment
2. detect logical blocs of comments based on indentation
3. move such comments to be pre-comments

In my opinion, the best visually is to move to pre-comments, but it would merge this comment with all pre-comments, but it's not *reversible*, ie if the line before the comment gets shorter, the next run of rustfmt will not place back the comment as post-comment. Probably not a big issue.

Option 2 is a bit (or lot ?) more work, for something that will anyway looks bad. I think it's fair to leave things in a non guidelines-conforming state when the places of the comments are weired, if we don't make things worse.
Option 1 being more conservative than option 3. I think you did it right. The decision to move comments around is probably better to discuss in another PR.